jena-text/src/test/java/org/apache/jena/query/text/TestTextIndexES.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jena.query.text;
+
+
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.vocabulary.RDFS;
+import org.elasticsearch.action.admin.indices.exists.indices.IndicesExistsRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+/**
+ *
+ * Integration test for {@link TextIndexES} class
+ * ES Integration test depends on security policies that may sometime not be loaded properly.
+ * If you find any issues regarding security set the following VM argument to resolve the issue:
+ * -Dtests.security.manager=false
+ *
+ */
+@ESIntegTestCase.ClusterScope()
+public class TestTextIndexES extends ESIntegTestCase {
```

Comment:
These are good unit tests, but they don't really exercise all the difficult cases. Could you use the Germany/Berlin example I gave earlier to construct some more unit tests?

For example, start with a triple `:de rdfs:label "Germany"@en` and test that the index finds the entity using `Germany`. Then add the label `"Deutschland"@de` to the same entity and test that it can be found using both labels. Then delete the `"Germany"@en` label and test that the entity can be found using `Deutschland` but not `Germany`.

Similar for the Berlin examples: Start with `:berlin rdfs:label "Berlin"@en` and test that the entity can be found using `Berlin`. Then add `"Berlin"@de` and test that it can still be found. Remove `"Berlin"@en` and test that the entity can still be found using `Berlin`. Finally remove `"Berlin"@de` and test that the entity can no longer be found using `Berlin`.